R/api.R

```diff
@@ -191,22 +205,37 @@ handleAPIfailure <- function(code, response) {
         return(crGET(response$url))
     }
     msg <- http_status(response)$message
+    msg2 <- NULL
     if (code == 404) {
         # Add the URL that was "not found" (there isn't going to be any
         # useful response content message)
         msg2 <- response$url
     } else {
-        msg2 <- try(content(response)$message, silent = TRUE)
+        err_content <- try(content(response), silent = TRUE)
+        if (!is.error(err_content) && is.list(err_content)) {
+            if (!is.null(err_content$message)) {
+                msg2 <- err_content$message
+            } else if (
+                !is.null(err_content$element) &&
+                err_content$element == "crunch:error" &&
+                !is.null(err_content$description)
+            ) {
+                msg2 <- err_content$description
+            }
+        }
     }
-    if (!is.error(msg2)) {
+
+    if (!is.null(msg2)) {
         msg <- paste(msg, msg2, sep = ": ")
```

Comment:
Random rant: I always found it annoying that paste includes the separator even with NULL. If `paste('foo', NULL, sep = ':')` returned 'foo', you wouldn't need an if statement in cases like this one.